### DISCUSSION TITLE
Corefinment non-manifold edge cases

### DISCUSSION THREAD

**Discussion by pentacular** [CGAL/cgal#8602] [Q&A]
I see "Non-manifold edge case 1~9" in Face_graph_output_builder.h.

Are these edge cases documented somewhere?

I'd like to understand what is going wrong so that I can fix the input. :)

Also, is it possible for us to add correction strategies to deal with these cases as they arise?

**Comment by sloriot** [comment]
These are cases when the output is non-manifold. In such case, you'd get false returned by the functions. Do you have typical cases where this happens and that you can share? The way to handle it also might depends on what you want to do with the output of the Boolean.

**Reply by pentacular** [reply]
Right, but I expect that it's non-manifold for reasons.

e.g., if there's a zero distance gap between corners or edges.

This can be corrected by separating the vertices slightly.

But this requires understanding the nature of the defect.

**Comment by sloriot** [comment]
This example: https://github.com/CGAL/cgal/blob/master/Polygon_mesh_processing/examples/Polygon_mesh_processing/corefinement_mesh_non_manifold_intersection.cpp shows the addition of a visitor to handle non-manifold output as a triangle soup. You can call `orient_polygon_soup()` and non-manifold features will be duplicated and you'll end up with self-intersections there. It still does not answer your question but the problem you will have is an edge in the output that will be non-manifold.

**Reply by pentacular** [reply]
Ah, I think I misunderstood what that visitor does, but I'm still a little puzzled.

If corefine_and_compute_intersection() returns false, it means that it is not manifold.
I assumed that non-manifold mesh was otherwise valid and could be fixed.
But if this is the case, what is the advantage of the Non_manifold_output_visitor?

I assume that either the non-manifold mesh is not valid, or there must be some reason not to just convert that into a polygon soup.

What am I missing?
